### PR TITLE
chore(ci/pr-checks): add lint checks

### PR DESCRIPTION
**Pull request by sidvishnoi**
## Context

Part of https://github.com/interledger/web-monetization-extension/issues/298

## Changes proposed in this pull request

Runs lint, format and typescheck in CI. Doesn't fail entire workflow if lint fails - for now.
Added as separate steps for clearer output.

### CODE REVIEW

**Comment by raducristianpopa** [comment]
<h2>Extension builds preview</h2>

<table role="table">
<thead>
  <tr>
    <th>Name</th>
    <th>Link</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td align="center">Latest commit</td>
    <td align="center">c691baef44df1518f6048ea4cffe587551a7237c</td>
  </tr>
  <tr>
    <td align="center">Latest job logs</td>
    <td align="center"><a href="https://github.com/interledger/web-monetization-extension/actions/runs/9383046937">Run #9383046937</a></td>
  </tr>
  <tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/24556012876/artifacts/1570647928">Download</a></td></tr><tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/24556012876/artifacts/1570647930">Download</a></td></tr>
</tbody>
</table>

File: .github/actions/setup/action.yml [hunk 1]
```diff
@@ -6,24 +6,22 @@ runs:
...
       with:
         run_install: false
 
-    - name: Get PNPM store path
-      id: pnpm-cache
+    - name: Get pnpm store directory
       shell: bash
       run: |
-        echo "pnpm_cache_dir=$(pnpm store path)" >> $GITHUB_OUTPUT
+        echo "PNPM_STORE_PATH=$(pnpm store path)" >> $GITHUB_ENV
```
Comment (raducristianpopa) [inline .github/actions/setup/action.yml:19]: What's the reason for putting this in the env instead of output?

File: package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
   "scripts": {
     "build": "tsx ./scripts/build.ts",
     "dev": "tsx ./scripts/dev.ts",
-    "lint": "concurrently \"pnpm:lint:*\"",
+    "lint": "concurrently \"pnpm:lint:*(!fix)\"",
```
Comment (raducristianpopa) [inline package.json:14]: I would suggest this renaming and update of the scripts to also take advantage of the cache:

```suggestion
    "lint": "eslint . --cache --cache-location 'node_modules/.cache/eslintcache' --max-warnings 0",
    "lint:fix": "eslint . --fix --cache --cache-location 'node_modules/.cache/eslintcache' --max-warnings 0",
     "format": "prettier . --check --cache --cache-location='node_modules/.cache/prettiercache'",
     "format:fix": "prettier . --write --cache --cache-location='node_modules/.cache/prettiercache' --log-level=warn",
     "typecheck": "tsc --noEmit"
```

**Comment by raducristianpopa** [comment]
> The failing lint check is for demonstration only. It's getting fixed in other PR.

The error needs to be fixed, otherwise merging is blocked.

File: .github/actions/setup/action.yml (hunk 1 above)
Comment (sidvishnoi) [reply .github/actions/setup/action.yml:19]: Copied what pnpm docs suggest latest. No other reason.

File: package.json (hunk 2 above)
Comment (sidvishnoi) [reply package.json:14]: Rename: Sure.

Oh nice. TIL they all added caches.
Let me check if CI caches node_modules or pnpm store...

**Comment by mkurapov** [comment]
> The error needs to be fixed, otherwise merging is blocked.

There doesn't seem to be any required checks to pass on this repo, just needs an approval. 

What should be required? @sidvishnoi @raducristianpopa 
If you don't have permissions, I can make those changes

**Comment by sidvishnoi** [comment]
I think we can require all checks except lint?

**Review by raducristianpopa — APPROVED** [review]

**Pull request merged**